controllers/openstackdataplanenode_controller.go

```diff
@@ -316,11 +316,13 @@ func (r *OpenStackDataPlaneNodeReconciler) GenerateInventory(ctx context.Context
 	all := inventory.AddGroup("all")
 	host := all.AddHost(instance.Name)
 
+	networkConfig := r.GetAnsibleNetworkConfig(instance, instanceRole)
+
 	host.Vars["ansible_user"] = r.GetAnsibleUser(instance, instanceRole)
 	host.Vars["ansible_port"] = r.GetAnsiblePort(instance, instanceRole)
 	host.Vars["managed"] = r.GetAnsibleManaged(instance, instanceRole)
 	host.Vars["management_network"] = r.GetAnsibleManagementNetwork(instance, instanceRole)
-	host.Vars["network_config"] = r.GetAnsibleNetworkConfig(instance, instanceRole)
+	host.Vars["network_config"] = deployment.NicConfigTemplateFile
```

Comment:
This won't work if we want to allow users to choose a default template actually. Might need to conditionally set this based on the value of `networkConfig`. Which is where it previously got its value from, only that the return value of that function now returns the actual template.

So if `networkConfig != ""`. Then we set this to `deployment.NicConfigTemplateFile` else, we don't set it?